### ISSUE TITLE
download internal hyperlinks

### ISSUE THREAD

**Issue by AlirezaMalih** [medialab/hyphe#498]
are there any access point or button to download all the internal hyperlinks in gexf file in a same time?
for instance, i have 100 urls to crawl, so for each of them i should go manually and download its own gexf file which includes their internal hyperlinks, so i wonder are there any access to download all these internal hyperlinks of all these 100 urls in a same time? 
tnx :)

**Comment by SeyedAlirezaMalih** [comment]
> I don't remember the UI being able to perform this, no. But this command line tool here (namely the `hyphe dump` command) should be able to do so: https://github.com/medialab/minet/blob/master/docs/cli.md#dump


tnx :), i tried to use it and download the csv file, but this file is not completed like gexf format, cause i should convert them to gexf format and it would not be like that format, specially when i want to use them with NetworkX, which would not be like the original format. can i download the as gexf format?

**Comment by Yomguithereal** [comment]
> can i download the as gexf format?

I don't think you can. But the CSV file should be very easy to load as a graph using networkx with some data wrangling. Else you can also try this tool: https://medialab.github.io/table2net/

**Comment by SeyedAlirezaMalih** [comment]
> > can i download the as gexf format?
> 
> I don't think you can. But the CSV file should be very easy to load as a graph using networkx with some data wrangling. Else you can also try this tool: https://medialab.github.io/table2net/

merci, well i tried to load the csv with networkx, but the main problem is the csv file have the sources link which crawled, but not the target sources which from a web page goes to other webpages, so its like just nodes of webpages but without the paths or targets. do you know how can i download or include them in csv file? tnx :)

**Comment by boogheta** [comment]
Hello there, I'm not sure I understand exactly your needs : are you trying to do a graph of links between webpages and not of websites ?
If that's the case, you should build a corpus with the setting default creation rule set to page in the Settings when creating the corpus, then the network should be made of what you're looking for.

**Comment by SeyedAlirezaMalih** [comment]
> Hello there, I'm not sure I understand exactly your needs : are you trying to do a graph of links between webpages and not of websites ? If that's the case, you should build a corpus with the setting default creation rule set to page in the Settings when creating the corpus, then the network should be made of what you're looking for. Otherwise, Hyphe is made to aggregate links between web entities that are groups of webpages, getting the detailed links between all the pages is possible but not trivial and would require you to call the API manually.


thank you, well i'm looking forward to download each by each these webpages corpus, internal hyperlinks corpus, for instance, i have a website which i crawled, so the internal network corpus within the website can be download manually as gexf format, i mean the internal hyperlinks can be downloadable by each of them, i need to download these gexf file each by each cause i have too many webpages too crawl and the process of downloading each gexf file takes time. Merci :)

**Comment by SeyedAlirezaMalih** [comment]
> Hello there, I'm not sure I understand exactly your needs : are you trying to do a graph of links between webpages and not of websites ? If that's the case, you should build a corpus with the setting default creation rule set to page in the Settings when creating the corpus, then the network should be made of what you're looking for. Otherwise, Hyphe is made to aggregate links between web entities that are groups of webpages, getting the detailed links between all the pages is possible but not trivial and would require you to call the API manually.

it might be better to mention like this:

i can't download internal hyperlinks, with minet also, i tried to put ignore_internal_links= True, but didnt work, i'm trying to download the internal hyperlinks and dots with 'gexf' format, but for large amount of data i cant manually do this.

**Comment by SeyedAlirezaMalih** [comment]
I mean the data inside of 'PagesNetwork' in I cant download them all, just can have the csv for nodes (pages), but not the hyperlinks between them.

**Comment by boogheta** [comment]
closing this since there was no more answer for a while. Please feel free to reopen if necessary
